lib/TreeSearch.js

```diff
@@ -5,50 +5,55 @@ const TreeSearch = {};
 // Returns a function that performs a preorder search on the tree for the given
 // simplification function
 TreeSearch.preOrder = function(simplificationFunction) {
-  return function (node) {
-    return search(simplificationFunction, node, true);
+  return function (node, scope={}) {
+    return search(simplificationFunction, node, true, scope);
   };
 };
 
 // Returns a function that performs a postorder search on the tree for the given
 // simplification function
 TreeSearch.postOrder = function(simplificationFunction) {
-  return function (node) {
-    return search(simplificationFunction, node, false);
+  return function (node, scope={}) {
+    return search(simplificationFunction, node, false, scope);
   };
 };
 
 // A helper function for performing a tree search with a function
-function search(simplificationFunction, node, preOrder) {
+function search(simplificationFunction, node, preOrder, scope={}) {
   let status;
 
   if (preOrder) {
-    status = simplificationFunction(node);
+    status = simplificationFunction(node, scope);
     if (status.hasChanged()) {
       return status;
     }
   }
 
-  if (Node.Type.isConstant(node) || Node.Type.isSymbol(node)) {
+  if (Node.Type.isConstant(node)) {
     return Node.Status.noChange(node);
   }
+  // Moved isUnaryMinus check above Symbol check to address symbol nested inside unaryminus
   else if (Node.Type.isUnaryMinus(node)) {
-    status = search(simplificationFunction, node.args[0], preOrder);
+    status = search(simplificationFunction, node.args[0], preOrder, scope);
     if (status.hasChanged()) {
       return Node.Status.childChanged(node, status);
     }
   }
+  // Break out isSymbol test and add a changeType for SUBSTITUTE_SYMBOL?
```

Comment:
don't need this comment (it's more like a commit message and less like a long term comment, like above)

Though if you want to keep a comment here it'd probably be more like `// symbols can be simplified through variable substitution`